### ISSUE TITLE
Expand Selection To Tag

### ISSUE THREAD

**Issue by philsinatra** [atom/atom#6008]
Are there any plans to add an option to Expand Selection To Tag, similar to a default option available in Sublime Text? I'm hoping to find out if it's in the future plans for Atom before spending time developing a package for this functionality.

**Comment by izuzak** [comment]
@philsinatra I think you're looking for this command: [`Bracket Matcher: Select Inside Brackets`](https://github.com/atom/bracket-matcher/blob/73979ad1660748e1b5c1419022ed8561b52263b1/keymaps/bracket-matcher.cson#L6)

Let me know if I misunderstood!

**Comment by philsinatra** [comment]
@izuzak Not exactly. I'm looking to select the content between opening/closing HTML tags:

```html
<p>Lorem ipsum dolor sit amet</p>
```

With the cursor anywhere between the opening and closing paragraph tags, a quick way to select only the _Lorem&hellip;_ text, not including the actual `<p>` and `</p>` tags.

If my code look liked this:

```html
<div class="wrapper">
  <div class="main">
    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Corrupti ipsam ullam magni minus, aperiam animi, quasi corporis. Impedit expedita, excepturi, iure corporis quo maxime, aperiam perferendis voluptas praesentium sint quam.</p>
  </div>
</div>
```

and I placed the cursor after `class="wrapper">` and performed this shortcut, I could quickly select all of the content within the `.wrapper` div.

This is basically the only major function I used constantly in Sublime Text that I can't find an equivalent for in Atom so far.

**Comment by philsinatra** [comment]
`Control + ,`

I just found it! Thanks.

**Comment by izuzak** [comment]
@philsinatra Strange, as far as I can tell, the command I suggested above does exactly that. Can you explain on that case what you expected to see selected and how that's different from what the command selected? A screenshot would be helpful here, I think, where you would point to the difference..

**Comment by izuzak** [comment]
> `Control + ,`
> I just found it! Thanks.

Which OS are you using? That keybinding doesn't trigger anything on OSX as far as I can tell. Can you open the keybinding resolver and take a screenshot of what command is invoked when you trigger that keybinding?

https://github.com/atom/keybinding-resolver

**Comment by izuzak** [comment]
Ahhh, I see it now (I was testing on a different example), it's definitely selecting more than it should:

![screen shot 2015-03-18 at 18 43 34](https://cloud.githubusercontent.com/assets/38924/6715430/b539cd64-cd9e-11e4-9492-35be8e3aaea7.png)

I _think_ that's a bug in the bracket matcher -- would you mind opening an issue over at https://github.com/atom/bracket-matcher/issues and provide all the details which demonstrate the problem?

**Comment by izuzak** [comment]
Nevermind, just created it: https://github.com/atom/bracket-matcher/issues/117. Sorry for the noise, and thanks for the example.

**Comment by philsinatra** [comment]
It looks like the functionality I'm tapping into is part of [Emmet](https://atom.io/packages/emmet)

![screen shot 2015-03-18 at 1 56 12 pm](https://cloud.githubusercontent.com/assets/1465808/6715705/a001e3fa-cd76-11e4-84b8-7bd5526e1c5f.png)

**Comment by izuzak** [comment]
Ahh, cool -- thanks for letting me know.

**Comment by philsinatra** [comment]
Thanks for the quick response and for looking into this. I have an additional question about one other function I'm looking for. I'll post a new issue with details.

**Comment by fricks** [comment]
not working anymore, please help!

**Comment by hex0gen** [comment]
Windows 10 + Atom 1.10.2 + [bracket-matcher](https://atom.io/packages/fancy-bracket-matcher) = love. So ctrl-alt-m keys combination makes a needed magic.
* P.S. This post for future generations, who searching for answer on this question

**Comment by jessephillips** [comment]
This still doesn't work on Atom. Or do I not have the right packages?

**Comment by Ben3eeE** [comment]
@jessephillips This should be fixed in Atom 1.22 currently in beta
